### PR TITLE
[10.x] Add Collection::select() method

### PR DESCRIPTION
**Pull request by morrislaptop**
This adds a new `Collection::select()` method that selects a certain number of keys from a multi-dimensional array. 

It's kind of like `only` except for arrays of arrays.

It's also like `pluck` except it supports multiple keys. 

The naming is inspired by SQL and the Query Builder. 

`$users->select('id', 'name', 'email'])` would be the equivalent of:

```php
$users = $users->map(function ($user) {
  return collect($user)->only(['id', 'name', 'email']);
});
```

### CODE REVIEW

**Comment by decadence** [comment]
Just yesterday I was wondering if there is such a method. And here it is.

File: src/Illuminate/Collections/Collection.php [hunk 1]
```diff
@@ -918,6 +918,27 @@ public function only($keys)
         return new static(Arr::only($this->items, $keys));
     }
 
+    /**
+     * Get the values for the specified keys from an array.
+     *
+     * @param  \Illuminate\Support\Enumerable<array-key, TKey>|array<array-key, TKey>|string|null  $keys
+     * @return static
+     */
+    public function select($keys)
```
Comment (driesvints) [inline src/Illuminate/Collections/Collection.php:927]: Could perhaps use argument unpacking here.

Comment (morrislaptop) [reply src/Illuminate/Collections/Collection.php:927]: I went for consistency with the `only` method... but happy to move to argument unpacking!

Comment (driesvints) [reply src/Illuminate/Collections/Collection.php:927]: All good 👍

**Review by Williamug — COMMENTED** [review]
:+1:

**Pull request merged**